Automattic/wp-calypso
# Add/new email comparison generic mail form

#### Changes proposed in this Pull Request

This pull request adds minimal code to show a working control modifying the underneath provider cards that will swap from annual billing to monthly billing.

This pull request also refactors, remove duplicated code, and cleans the previous code. Please, pair with me if you need help on reviewing this PR, as it became very big.

#### Testing instructions

1. Go to a site with domains without email subscription *
2. Click on 'Add email' for the desired site
3. Once you are in the comparison page, add at the end the following flag: `?flags=emails/new-email-comparison`
4. You should be able to see a billing selector (pay annually/monthly) that triggers changes in the underneath cards.
5. Remove from cart the previous item and add the other email provider
6. Play around with the validations (Add another mailbox is enabled in Google Workspace Card)

**Remarks**
You can also go through the path /email/{domain}/purchase/{site}?flags=emails/new-email-comparison
This change the product type yet, but products are not ready yet?.

![image](https://user-images.githubusercontent.com/5689927/144469509-7e907e1c-c491-4289-a998-59cf21732233.png)

Related to {1200182182542585-as-1201464738311301}
